dev/CommonStyles/AppBarSeparator.xaml

```diff
@@ -0,0 +1,70 @@
+                    <Grid x:Name="RootGrid">
+                        <VisualStateManager.VisualStateGroups>
+                            <VisualStateGroup x:Name="ApplicationViewStates">
+                                <!-- FullSize is used when we are in landscape or filled mode -->
+                                <VisualState x:Name="FullSize">
+                                    <VisualState.Setters>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <!-- Compact is used when we are in portrait or snapped mode -->
+                                <VisualState x:Name="Compact">
+                                    <VisualState.Setters>
+                                        <Setter Target="RootGrid.Height" Value="{ThemeResource AppBarThemeCompactHeight}"/>
+                                        <Setter Target="RootGrid.VerticalAlignment" Value="Top"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Overflow">
+                                    <Storyboard>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SeparatorRectangle" Storyboard.TargetProperty="Width">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="NaN" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SeparatorRectangle" Storyboard.TargetProperty="HorizontalAlignment">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="Stretch" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SeparatorRectangle" Storyboard.TargetProperty="Height">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="1" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SeparatorRectangle" Storyboard.TargetProperty="Margin">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="12,4,12,4" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                    </Storyboard>
+                                </VisualState>
+
+                            </VisualStateGroup>
+
+                        </VisualStateManager.VisualStateGroups>
+                        <Rectangle x:Name="SeparatorRectangle"
+                            Width="1"
+                            VerticalAlignment="Stretch"
+                            Fill="{TemplateBinding Foreground}"
+                            Margin="16,12,15,12"/>
```

Comment:
How about introducing a theme resource for this margin as we do for the [NavigationView's separator](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/NavigationView/NavigationView.xaml#L777)? Proposed naming: `AppBarSeparatorMargin`.

Reply:
Generally, you shouldn't introduce new APIs or changes in a PR just because you happen to touch a specific area or file. However in this case, I think it's okay since we are introducing a somewhat new style here so I added the lightweight styling resources.

Reply:
`Generally, you shouldn't introduce new APIs or changes in a PR just because you happen to touch a specific area or file.` 

Agreed, I mainly used this occasion to bring this to our attention (aka "while I'm looking at this, I noticed..."). Would have been fine if that were addressed in a follow-up PR as well.